### ISSUE TITLE
Disambiguation

### ISSUE THREAD

**Issue by ziodave** [wetneb/opentapioca#33]
Hello @wetneb,

Can the following be improved or fixed by running the classifier on a larger dataset? or is it only based on the number of statements for each entity?

![image](https://user-images.githubusercontent.com/11438/110820805-ba273d00-828f-11eb-9727-e703893517ab.png)

![image](https://user-images.githubusercontent.com/11438/110820832-beebf100-828f-11eb-85a9-6aa8925183ee.png)

Cheers,
David

**Comment by wetneb** [comment]
It's hard to say looking at a single example! In general I would recommend training a model for a specific domain (which would also let you select more finely which entities can be annotated).

**Comment by ziodave** [comment]
@wetneb is there a reason you're not using the _edges_ in OpenTapioca/Wikidata to run the classifier? Did you try already?

**Comment by wetneb** [comment]
How would you turn them into features?

**Comment by ziodave** [comment]
I am not sure, I am trying to understand how to solve the above reference case.

I can see the Apple Inc. (Q312) has an edge to Steve Jobs (Q19837) and viceversa. I am trying to understand if we can use these edges to perform the training.

Because the training is extremely complex on larger datasets, involving first the dbpedia.org to wikidata.org item id mapping and then the classification. And I am afraid that SVM is not suitable for classification of large datasets.

**Comment by wetneb** [comment]
Yes edges are used, not directly as features of the classifier, but to foster the score of collections of mentions that are interlinked. It's something I tried to explain in the paper but things might well be unclear there, let me know if I can clarify aspects of that.

**Comment by ziodave** [comment]
I see. I think there some tuning to do then, take for example this scenario:

"Apple is a fruit", works nicely:

![image](https://user-images.githubusercontent.com/11438/112032763-05b5d280-8b3d-11eb-8e70-1550e65a0169.png)
![image](https://user-images.githubusercontent.com/11438/112032778-09e1f000-8b3d-11eb-9cfd-602fef7bc1a4.png)


"Apple Inc has been founded by Steve Jobs", works nicely:

![image](https://user-images.githubusercontent.com/11438/112032708-f6368980-8b3c-11eb-9178-4264115f7a60.png)
![image](https://user-images.githubusercontent.com/11438/112032729-fafb3d80-8b3c-11eb-8d9b-1310b1d1c701.png)


but "Apple has been founded by Steve Jobs", doesn't work nicely:

![image](https://user-images.githubusercontent.com/11438/112032874-21b97400-8b3d-11eb-8862-e4f5d716cc01.png)
![image](https://user-images.githubusercontent.com/11438/112032908-2a11af00-8b3d-11eb-9010-7a25ed79b1bb.png)


The only thing that's changing is that "Apple" in the second case is an alias of "Apple Inc.". As far as I understand from the paper though aliases have the same score as labels? Should the "Apple has been founded by Steve Jobs" actually yield the same results as "Apple Inc has been founded by Steve Jobs"?

**Comment by wetneb** [comment]
Yes it's a curious result, I would definitely expect "Apple has been founded by Steve Jobs" to be understood correctly. I think the training set probably did not contain extremely famous companies or people like those, so they are outliers that aren't scored well by the classifier.

**Comment by ziodave** [comment]
Yes, and this is the path that I followed, I thought I could find a better dataset for training, but this increases complexity. And then I thought is there a way to optimize cases similar to this one? at the end 'Apple' is just an alias of 'Apple Inc.' :-)

**Comment by wetneb** [comment]
I don't know exactly what you mean by optimize cases similar to this one - do you mean tweaking the classifier parameters manually to adjust for those cases?

**Comment by ziodave** [comment]
Yes, sorry. But is this a classifier issue? "Apple Inc" is detected while "Apple" as alias isn't.

Sorry, I might not be to expert with the classifier. I am asking a colleague of mine to join this thread too.

**Comment by wetneb** [comment]
"Apple Inc" is more specific than "Apple" so it is normal that is detected more easily.

**Comment by sareaghaei** [comment]
> I don't know exactly what you mean by optimize cases similar to this one - do you mean tweaking the classifier parameters manually to adjust for those cases?

I think the reason of not working fine for some cases is not necessarily classification. Changing the parameters may further reduce the evaluation metrics and as a result, be less accurate about things that already work well.
